### PR TITLE
[Php55] Make StringClassNameToClassConstantRector configurable to keep first pre-backslash string configurable

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba @leoloso @staabm @kenjis per discussion on https://github.com/rectorphp/rector-src/pull/5338#pullrequestreview-1772199580 , this PR propose to make `StringClassNameToClassConstantRector` configurable to allow set pre-backslash removed, but keep it as default.

I am open if it will need to be removed by default, but make it at least configurable so user can configure when needed.

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba @leoloso @staabm @kenjis it is ready for review.

**Comment by kenjis** [comment]
The distinction between `'\Foo'` and `'Foo'` as a classname is not syntactically correct in PHP.
However, in reality there are cases where these strings are distinguished, so it is better to have this option.
I don't care if the default is to remove the back slash.

File: rules/Php55/Rector/String_/StringClassNameToClassConstantRector.php [hunk 1]
```diff
@@ -37,6 +38,13 @@ final class StringClassNameToClassConstantRector extends AbstractRector implemen
      */
     private array $classesToSkip = [];
 
+    private bool $isKeepFirstBackslashString = true;
+
```
Comment (TomasVotruba) [inline rules/Php55/Rector/String_/StringClassNameToClassConstantRector.php:42]: This should be false by default.

It seems a bit verbose, what about "shouldKeepPreslash"?

Comment (samsonasik) [reply rules/Php55/Rector/String_/StringClassNameToClassConstantRector.php:42]: sure, updated 👍 https://github.com/rectorphp/rector-src/pull/5354/commits/ed70ee9467ffbda5c0c35e01bf5728578bd94c01

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you, looks great :+1:

**Pull request merged**